ib_cicd/promote_sb_solution.py

```diff
@@ -0,0 +1,291 @@
+    compile_solution,
+)
+from ib_cicd.migration_helpers import (
+    download_ibsolution,
+    download_dependencies_from_dev_and_upload_to_prod,
+    publish_dependencies,
+)
+
+from ib_cicd.promote_solution import (
+    version_tuple,
+    get_latest_ibsolution_path,
+    set_output_version_github,
+)
+from dotenv import load_dotenv
+
+load_dotenv()
+
+TARGET_IB_API_TOKEN = os.environ.get("TARGET_IB_API_TOKEN")
+SOURCE_IB_API_TOKEN = os.environ.get("SOURCE_IB_API_TOKEN")
+TARGET_IB_HOST = os.environ.get("TARGET_IB_HOST")
+SOURCE_IB_HOST = os.environ.get("SOURCE_IB_HOST")
+SOLUTION_BUILDER_NAME = os.environ.get("SOLUTION_BUILDER_NAME")
+FLOW_NAME = os.environ.get("FLOW_NAME")
+WORKSPACE_DRIVE_PATH = os.environ.get("WORKSPACE_DRIVE_PATH")
+TARGET_IB_PATH = os.environ.get("TARGET_IB_PATH")
+SOURCE_WORKING_DIR = os.environ.get("SOURCE_WORKING_DIR")
+DEPENDENCIES = os.environ.get("DEPENDENCIES")
+
+
+def get_latest_flow_version(flow_path, ib_host, ib_token) -> str:
+    """
+    Iterates through the files in a folder to get the filename with the maximum version number
+    :param flow_path: (string) IB path to directory where versioned flows are stored
+    :param ib_host: (string) IB host URL (e.g. https://platform.instabase.com/)
+    :param ib_token: (string) API token for IB environment
+    :return: (string) latest version in directory in <major>.<minor>.<patch> format
+    """
+    latest_version = "0.0.0"
+    try:
+        paths = list_directory(ib_host, flow_path, ib_token)
+    except Exception:
+        return latest_version
+
+    for path in paths:
+        p = pathlib.Path(path)
+        if re.fullmatch("[0-9]+.[0-9]+.[0-9]+", p.stem):
+            if version_tuple(p.stem) > version_tuple(latest_version):
+                latest_version = p.stem
+    return latest_version
+
+
+def get_sb_flow_path(solution_builder_name, flow_name, ib_root, ib_host, ib_token):
+    """
+    Iterates through flows in a solution builder project to return the path to the input flow name
+    :param solution_builder_name: (string) name of solution builder project
+    :param flow_name: (string) name of the flow
+    :param ib_root: (string) path to the IB drive (e.g. hannahroiter/ci-cd/fs/Instabase Drive)
+    :param ib_host: (string) IB host URL (e.g. https://platform.instabase.com/)
+    :param ib_token: (string) API token for IB environment
+    :return: (string) IB filesystem path to flow version
+    """
+    flows_path = os.path.join(
+        ib_root, ".instabase_projects", solution_builder_name, "latest", "flows"
+    )
+    paths = list_directory(ib_host, flows_path, ib_token)
+    for path in paths:
+        read_response = read_file_through_api(
+            ib_host, ib_token, os.path.join(path, "metadata.json")
+        ).content
+        metadata = json.loads(read_response)
+        if metadata["name"] == flow_name:
+            version = list_directory(ib_host, os.path.join(path, "versions"), ib_token)[
+                0
+            ]
```

Comment:
```suggestion
            version = list_directory(ib_host, os.path.join(path, "versions"), ib_token)[0]
```
Is there a reason we take the first item from the list_directory output (is there always only one)?